spec/uploader_spec.rb

```diff
@@ -493,19 +498,57 @@
-    io.puts(header)
-    5880000.times{ io.write("\xFF") }
-    io.rewind
-    result = Cloudinary::Uploader.upload_large(io, :chunk_size => 5243000, :tags => [TEST_TAG, TIMESTAMP_TAG])
-    expect(result["resource_type"]).to eq('raw')
-    io.rewind
-    result = Cloudinary::Uploader.upload_large(io, :resource_type => 'image', :chunk_size => 5243000, :tags => [TEST_TAG, TIMESTAMP_TAG])
-    expect(result["resource_type"]).to eq('image')
-    expect(result["width"]).to eq(1400)
-    expect(result["height"]).to eq(1400)
-    expect(result["format"]).to eq("bmp")
+    filename = "#{UNIQUE_TEST_ID}_cld_upload_large"
+
+    Tempfile.open([filename, ".bpm"]) do |temp_file|
+      Cloudinary.populate_large_file(temp_file, LARGE_FILE_SIZE)
+
+      temp_file_name = temp_file.path
+      temp_file_filename = File.basename(temp_file).split('.').first
+
+      expect(File.size(temp_file_name)).to eq(LARGE_FILE_SIZE)
+
+      resource = Cloudinary::Uploader.upload_large(temp_file_name,
+                                                   :chunk_size => LARGE_CHUNK_SIZE,
+                                                   :tags => ["upload_large_tag", TIMESTAMP_TAG],
+                                                   :filename => temp_file_filename)
+
+      expect(resource["tags"]).to eq(["upload_large_tag", TIMESTAMP_TAG])
+      expect(resource["resource_type"]).to eq("raw")
+      expect(resource["original_filename"]).to eq(temp_file_filename)
+
+      resource2 = Cloudinary::Uploader.upload_large(temp_file_name,
+                                                    :chunk_size => LARGE_CHUNK_SIZE,
+                                                    :tags => ["upload_large_tag", TIMESTAMP_TAG],
+                                                    :resource_type => "image",
+                                                    :use_filename => true,
+                                                    :unique_filename => false,
+                                                    :filename => filename)
+
+      expect(resource2["tags"]).to eq(["upload_large_tag", TIMESTAMP_TAG])
+      expect(resource2["resource_type"]).to eq("image")
+      expect(resource2["original_filename"]).to eq(filename)
+      expect(resource2["original_filename"]).to eq(resource2["public_id"])
+      expect(resource2["width"]).to eq(LARGE_FILE_WIDTH)
+      expect(resource2["height"]).to eq(LARGE_FILE_HEIGHT)
+
+      resource3 = Cloudinary::Uploader.upload_large(temp_file_name,
```

Comment:
@YomesInc, let's try to avoid performing upload large tests too many times, it is time and traffic consuming. Please try to combine tests and reduce the number of calls, or use mocks.

Reply:
> @YomesInc, let's try to avoid performing upload large tests too many times, it is time and traffic consuming. Please try to combine tests and reduce the number of calls, or use mocks.

@const-cloudinary Not sure that you mean resource3 only or all calls to Cloudinary::Uploader.upload_large. Now there is a single real call only. Please look at this.